src/shared/libfido2-util.c

```diff
@@ -866,7 +866,7 @@ int fido2_generate_hmac_hash(
                                 .hup_fd = -EBADF,
                         };
 
-                        r = ask_password_auto(&req, /* flags= */ 0, &pin);
+                        r = ask_password_auto(&req, ASK_PASSWORD_ACCEPT_CACHED, &pin);
```

Comment:
this is not right, because this would mean even if the pin was wrong the first time we'd immediately use it again, because cached. And tha would then immediately trigger lockout because we retry in a tight loop...

Hence, this is OK to do, but only in the first iteration of the loop. i.e. add some boolean that is initially false, and you set to true after the first iteration and that you use to initialize the flags parameter from.